dataikuapi/dss/ml.py

```diff
@@ -1490,13 +1512,471 @@ def __init__(self, raw_settings, hyperparameter_search_params):
+        "SEASONAL_NAIVE": PredictionAlgorithmMeta("seasonal_naive_timeseries", SeasonalNaiveSettings),
+        "AUTO_ARIMA": PredictionAlgorithmMeta("autoarima_timeseries", AutoArimaSettings),
+        "SEASONAL_LOESS": PredictionAlgorithmMeta("seasonal_loess_timeseries", SeasonalLoessSettings),
+        "GLUONTS_NPTS_FORECASTER": PredictionAlgorithmMeta("gluonts_npts_timeseries", GluonTSNPTSForecasterSettings),
+        "GLUONTS_SIMPLE_FEEDFORWARD": PredictionAlgorithmMeta("gluonts_simple_feed_forward_timeseries", GluonTSSimpleFeedForwardSettings),
+        "GLUONTS_DEEPAR": PredictionAlgorithmMeta("gluonts_deepar_timeseries", GluonTSDeepARSettings),
+        "GLUONTS_TRANSFORMER": PredictionAlgorithmMeta("gluonts_transformer_timeseries", GluonTSTransformerSettings),
+        "GLUONTS_MQCNN": PredictionAlgorithmMeta("gluonts_mqcnn_timeseries", GluonTSMQCNNSettings),
+    }
+
+    TIMEUNITS = {"MILLISECOND", "SECOND", "MINUTE", "HOUR", "DAY", "BUSINESS_DAY", "WEEK", "MONTH", "QUARTER", "HALF_YEAR", "YEAR"}
+    INTERPOLATION_METHODS = {"NEAREST", "PREVIOUS", "NEXT", "LINEAR", "QUADRATIC", "CUBIC", "CONSTANT"}
+    EXTRAPOLATION_METHODS = {"PREVIOUS_NEXT", "NO_EXTRAPOLATION", "CONSTANT", "LINEAR", "QUADRATIC", "CUBIC"}
+    CATEGORICAL_IMPUTATION_METHODS = {"MOST_COMMON", "NULL", "CONSTANT", "PREVIOUS_NEXT", "PREVIOUS", "NEXT"}
+    DUPLICATE_TIMESTAMPS_HANDLING_METHODS = {"FAIL_IF_CONFLICTING", "DROP_IF_CONFLICTING", "MEAN_MODE"}
+
+    class PredictionTypes:
+        TIMESERIES_FORECAST = "TIMESERIES_FORECAST"
+
+    def __init__(self, client, project_key, analysis_id, mltask_id, mltask_settings):
+        AbstractTabularPredictionMLTaskSettings.__init__(self, client, project_key, analysis_id, mltask_id, mltask_settings)
+
+        prediction_type = self.get_prediction_type()
+        if prediction_type != self.PredictionTypes.TIMESERIES_FORECAST:
+            raise ValueError("Unknown prediction type: {}".format(prediction_type))
+
+    def get_external_feature(self, feature_name):
+        """
+        Gets the feature preprocessing params for a particular external feature. This returns a reference to the
+        feature's settings, not a copy, so changes made to the returned object will be reflected when saving
+
+        :return: A dict of the preprocessing settings for an external feature
+        :rtype: dict
+        """
+        return self.get_feature_preprocessing(feature_name)
+
+    def get_timestep_params(self):
+        """
+        Gets the time step parameters for the time series forecasting task. This returns a reference to the
+        time step parameters, not a copy, so changes made to the returned object will be reflected when saving
+
+        :return: A dict of the time step parameters
+        :rtype: dict
+        """
+        return self.mltask_settings["timestepParams"]
+
+    def set_timestep_params(self, timeunit=None, n_timeunits=None, end_of_week_day=None):
```

Comment:
If `get_timestep_params()` returns a dict, maybe this should simply be `set_timestep()`